### PR TITLE
Gmeding/AWS-CDK

### PR DESCRIPTION
**Pull request by Ghmeding**
**This PR:**

Adds a folder with a aws-cdk for deploying our database to AWS.
In simple terms, it just creates a postgresSQL database in AWS in frankfurt (eu-central-1).

See the ReadMe for information on how to connect.

### CODE REVIEW

File: aws-cdk/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+   - Username
+   - Password
+   - Hostname
+   - Port
+   - Database name
+
+3. **Connect using psql**:
+   Use the following command structure to connect:
+```
+psql -h waectdatabase-waectresultstoragedbe1ccf3e0-tmed4lgituj0.c10ia6ywc903.eu-central-1.rds.amazonaws.com -p 5432 -U waect -d waectResultStorage
```
Comment (dudustri) [inline aws-cdk/README.md:33]: I think it is better delete this guy from here since the repository is public, no?

File: aws-cdk/lib/rds-stack.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    const vpc = Vpc.fromLookup(this, 'waectResultStorageVPC', { vpcId: "vpc-08d145893bcbe80f7" });
+
+    // create a security group
+    const dbSg: cdk.aws_ec2.SecurityGroup = new SecurityGroup(this, "waectResultStorageSG", {
+      securityGroupName: "waectResultStorageSG",
+      vpc: vpc
+    });
+
+    // Allow incoming traffic from any IP address on port 5432
+    dbSg.addIngressRule(
```
Comment (dudustri) [inline aws-cdk/lib/rds-stack.ts:41]: nice

File: aws-cdk/lib/rds-stack.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      instanceType,
+      engine,
+      port,
+      securityGroups: [dbSg],
+      databaseName: dbName,
+      credentials: credentials,
+      backupRetention: Duration.days(0),
+      deleteAutomatedBackups: true,
+      removalPolicy: RemovalPolicy.DESTROY,
+      allocatedStorage: 20
```
Comment (dudustri) [inline aws-cdk/lib/rds-stack.ts:66]: is it in gb?

File: aws-cdk/lib/rds-stack.ts
Comment (dudustri) [inline aws-cdk/lib/rds-stack.ts:1]: good stuff

File: aws-cdk/package.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  "name": "aws-cdk",
+  "version": "0.1.0",
+  "bin": {
+    "aws-cdk": "bin/aws-cdk.js"
+  },
+  "scripts": {
+    "build": "tsc",
+    "watch": "tsc -w",
+    "test": "jest",
+    "cdk": "cdk"
```
Comment (dudustri) [inline aws-cdk/package.json:11]: Is this cdk that apply the infrastructure as code in the rds-stack file?

**Review by dudustri — APPROVED** [review]
Super nice Infra as code hehehe I have one question. What is the difference between using the the cdk and terraform?

File: aws-cdk/README.md (hunk 1 above)
Comment (Ghmeding) [reply aws-cdk/README.md:33]: Sure

File: aws-cdk/lib/rds-stack.ts
Comment (Ghmeding) [reply aws-cdk/lib/rds-stack.ts:1]: thanks bra

File: aws-cdk/lib/rds-stack.ts (hunk 2 above)
Comment (Ghmeding) [reply aws-cdk/lib/rds-stack.ts:41]: <33

File: aws-cdk/lib/rds-stack.ts (hunk 3 above)
Comment (Ghmeding) [reply aws-cdk/lib/rds-stack.ts:66]: yes

File: aws-cdk/package.json (hunk 4 above)
Comment (Ghmeding) [reply aws-cdk/package.json:11]: yes to deploy you 1) build, 2) maybe run cdk diff and then 3) cdk deploy to deploy the infrastructure

File: aws-cdk/package.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  "devDependencies": {
+    "@types/jest": "^29.5.12",
+    "@types/node": "22.5.4",
+    "jest": "^29.7.0",
+    "ts-jest": "^29.2.5",
```
[the remark this answers is not in the document]
Comment (Ghmeding) [reply aws-cdk/package.json:17]: yes good point

File: aws-cdk/cdk.context.json
[the remark this answers is not in the document]
Comment (Ghmeding) [reply aws-cdk/cdk.context.json:1]: yes, it basically just a cache of information received when deploying or running cdk synth

File: aws-cdk/jest.config.js
[the remark this answers is not in the document]
Comment (Ghmeding) [reply aws-cdk/jest.config.js:1]: done! :)

**Review by simoonH — APPROVED** [review]
Very sexy! Tested it with python-flask and AWS credentials works as expected, noice!

**Pull request merged**